snuba/clickhouse/formatter/expression.py

```diff
@@ -154,4 +163,59 @@ def visit_argument(self, exp: Argument) -> str:
     def visit_lambda(self, exp: Lambda) -> str:
         parameters = [self.__escape_identifier_enforce(v) for v in exp.parameters]
         ret = f"({', '.join(parameters)} -> {exp.transformation.accept(self)})"
-        return self.__alias(ret, exp.alias)
+        return self._alias(ret, exp.alias)
+
+
+class ClickhouseExpressionFormatter(ClickhouseExpressionFormatterBase):
```

Comment:
i'm wondering if it would be better to separate out these strategy / template methods into their own class that the `ClickhouseExpressionFormatter` could then delegate to

Reply:
I'd be ok with that as an alternative to the template method, but I suspect it would not age. 
This is the first iteration, then we will try to get smarter on what to remove and what not:
For example we will tweak the function_call formatter method probably to avoid redacting the literal in conditions like `project_id = 1`. I suspect it will become harder to build an interface decoupled enough from the formatter to have a meaningful interface.

Reply:
👍🏼